### PR TITLE
feat: add image tag expiration env var

### PR DESCRIPTION
**Pull request by tnevrlka**
add image tag expiration env var, see https://github.com/redhat-appstudio/e2e-tests/pull/548

### CODE REVIEW

File: components/build-service/development/image-expiration-patch.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: image-expiration
+  namespace: system
```
Comment (Michkov) [inline components/build-service/development/image-expiration-patch.yaml:5]: namespace not needed

File: components/build-service/development/image-expiration-patch.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: image-expiration
```
Comment (Michkov) [inline components/build-service/development/image-expiration-patch.yaml:4]: name of deployment is `build-service-controller-manager`

File: components/build-service/development/kustomization.yaml [hunk 3]
```diff
@@ -10,3 +10,9 @@ patches:
       kind: Deployment
       name: controller-manager
     path: logs-in-console-format-cmd-arg-patch.yaml
+  - target:
+      group: apps
+      version: v1
+      kind: Deployment
+      name: image-expiration
```
Comment (Michkov) [inline components/build-service/development/kustomization.yaml:17]: name of deployment is `build-service-controller-manager`

File: hack/preview.sh [hunk 4]
```diff
@@ -161,6 +161,7 @@ sed -i.bak "s/rekor-server.enterprise-contract-service.svc/$rekor_server/" $ROOT
 
 [ -n "${BUILD_SERVICE_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/build-service\")) |=.newName=\"${BUILD_SERVICE_IMAGE_REPO}\"" $ROOT/components/build-service/base/kustomization.yaml
 [ -n "${BUILD_SERVICE_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/build-service\")) |=.newTag=\"${BUILD_SERVICE_IMAGE_TAG}\"" $ROOT/components/build-service/base/kustomization.yaml
+[ -n "${BUILD_SERVICE_IMAGE_TAG_EXPIRATION}" ] && yq -i e "(.spec.template.spec.containers[].env[] | select(.name==\"BUILD_SERVICE_IMAGE_TAG_EXPIRATION\")) |= \"${BUILD_SERVICE_IMAGE_TAG_EXPIRATION}\"" $ROOT/components/build-service/development/kustomization.yaml
```
Comment (Michkov) [inline hack/preview.sh:164]: you want to modify `components/build-service/development/image-expiration-patch.yaml` and also you need to modify `.value` , so it should be:
```
[ -n "${BUILD_SERVICE_IMAGE_TAG_EXPIRATION}" ] && yq -i e "(.spec.template.spec.containers[].env[] | select(.name==\"IMAGE_TAG_EXPIRATION\") | .value) |= \"${BUILD_SERVICE_IMAGE_TAG_EXPIRATION}\"" $ROOT/components/build-service/development/image-expiration-patch.yaml
```

File: components/build-service/development/image-expiration-patch.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+      - name: manager
+        env:
+          - name: BUILD_SERVICE_IMAGE_TAG_EXPIRATION
```
Comment (Michkov) [inline components/build-service/development/image-expiration-patch.yaml:12]: The environment variable is `IMAGE_TAG_ON_PR_EXPIRATION`

**Review by psturc — APPROVED** [review]
/lgtm

**Review by psturc — APPROVED** [review]
/lgtm

**Pull request merged**